index.d.ts

```diff
@@ -650,13 +654,17 @@ export interface WithT {
   t: TFunction;
 }
 
+/**
+ * Gets the defined translation keys if they have been overridden, 'string' type otherwise
+ */
+type TKeys = TranslationKeys['keys'] extends string ? TranslationKeys['keys'] : string;
+                     
 export type TFunctionResult = string | object | Array<string | object> | undefined | null;
 export type TFunctionKeys = string | TemplateStringsArray;
 export interface TFunction {
   // basic usage
   <
     TResult extends TFunctionResult = string,
-    TKeys extends TFunctionKeys = string,
```

Comment:
`TFunctionKeys` is `string | TemplateStringsArray`. What was the point of `TemplateStringsArray`?

I guess I introduced a regression by removing this but I need to understand the type to fix it 😉